kasa/discover.py

```diff
@@ -358,12 +360,19 @@ async def connect_single(
         The device type is discovered by querying the device.
 
         :param host: Hostname of device to query
+        :param port: Optionally set a different port for the device
+        :param credentials: Optionally provide credentials for
+            devices requiring them
+        :param port: Optionally provide an id identifying the protocol
+            to use.  1 is for legacy/original, 2 is for klap
         :rtype: SmartDevice
         :return: Object for querying/controlling found device.
         """
         unknown_dev = SmartDevice(
             host=host, port=port, credentials=credentials, timeout=timeout
         )
+        if isinstance(protocol_class, TPLinkKlap):
+            unknown_dev.protocol = TPLinkKlap(host, credentials)
```

Comment:
```suggestion
        if protocol_class is not None:
            unknown_dev.protocol = protocol_class(host, credentials=credentials)
```
This will make it generic for future uses :-)

Reply:
Done